### DISCUSSION TITLE
[SOLVED] Filter by keyword (+/-)

### DISCUSSION THREAD

**Discussion by ThePowerTool** [martinrotter/rssguard#752] [Q&A] [answered]
Hi All!  1st post!

Would it be possible to filter a feed by +keyword-A -keyword-B (or more)?

I've been using RSS Guard and would like to use it more often.  My key requirement to quickly review feeds it so search and exclude keywords in a manner similar to grep (grep -i keyword-a | grep -vi keyword-b | etc.)

I believe, from what I've read here, that regex would also be supported which would be a big plus IMHO.

Thank you very much, in-advance!

**Comment by martinrotter** [comment]
Hi. Sorry for late reply, i am very very busy these days (small baby in the family).

**Comment by martinrotter** [comment]
> +keyword-A -keyword-B

What this means? You mean you only want to include articles which "contain" keyword-A and "DO NOT contain" keyword-B in the same time?

EDIT: And also, do you only want to "quick filter" existing messages or filter messages as they come in and drop/reject those not meeting your conditions?

**Comment by ThePowerTool** [comment, marked as answer]
Thank you very much for your time!  Congratulations on the baby!

Yes, that's why I provided the grep example (so, within the list of articles, the following would be a single filter of multiple keywords):
+keyword-A means items containing keyword-A and
 -keyword-B means items that do not include keyword-B
this could continue with
+keyword-C
-Keyword-D
* meaning all keywords applied to the list of articles

The way you described "quick filter" is what I want and important to me--I need the messages to remain regardless of how many times I filter so each filter starts with the entire set (or list) of articles.
Also "keyword" isn't necessarily a single word--it could be multiple words within quotes--
e.g. +"stormy weather" -"clear skies" or regex +"[S,s]tormy.weather" -"[C,c]lear.[S,s]kies"

Thank you, again, for your help!

**Reply by martinrotter** [reply]
In that case, there is "quick filter" box above article list and the box supports regexes.

This regex does what you want: `^(?=.*(?:opera))(?!.*(?:nederland)).*$`. This particular one requires articles to have "opera" and must not have "nederland".

This is what you want?


![image](https://user-images.githubusercontent.com/1255302/173294623-d190af8c-d281-422d-9eb0-5537609d6529.png)

**Reply by martinrotter** [reply]
Regexpes support is very extensive, basically is pcre-conformant.

https://doc.qt.io/qt-5/qregularexpression.html#details

**Comment by ThePowerTool** [comment]
I am embarrassed.  I should have just tried regex.

That is exactly what I need.

Thank you SO very much for your help!

Sincerely,

ThePowerTool
